tests/test_representation_manager.cc

```diff
@@ -178,87 +179,170 @@ namespace rascal {
-    auto cutoff_it{cutoffs.begin()};
+    const auto & data{ref_data.at("rep_info").template get<json>()};
+    // feature_matrices = data["feature_matrices"];
 
+    size_t manager_i{0};
     for (auto & manager : managers) {
-      if (verbose) {
-        if (filename_it != filenames.end()) {
-          std::cout << "Structure: " << *filename_it;
-          std::cout << " with cutoff " << *cutoff_it << std::endl;
-          ++cutoff_it;
-          if (cutoff_it == cutoffs.end()) {
-            cutoff_it = cutoffs.begin();
-            ++filename_it;
-          }
-        } else {
-          std::cout << "Structure: Filename unknown" << std::endl;
-        }
-      }
-      for (const auto & hyper : hypers) {
-        representations.emplace_back(manager, hyper);
-        // Should be done automatically in compute()
-        // TODO(max-veit) make that its own test case
-        // representations.back().precompute();
+      for (const auto & config : data.at(manager_i)) {
+        const auto & hypers = config.at("hypers").template get<json>();
+        const auto & ref_representation =
+            config.at("feature_matrix").template get<Std2DArray_t>();
+
+        representations.emplace_back(manager, hypers);
         representations.back().compute();
-        // Check dimensions of the storage array
-        size_t max_radial = hyper.at("max_radial");
-        size_t max_angular = hyper.at("max_angular");
-        BOOST_CHECK(representations.back().get_feature_size() ==
-                    max_radial * (max_angular + 1) * (max_angular + 1));
-        if (verbose) {
-          size_t center_idx{0};
-          for (auto center : manager) {
-            center_idx += 1;
-            std::cout << "Soap vector for center: " << center_idx;
-            std::cout << std::endl;
-            representations.back().print_soap_vector(center, std::cout);
+
+        // TODO(felix) quick fix of something that will disappear soon
+        FeatureManagerBlockSparse<double> features{
+            representations.back().get_feature_size(), hypers};
+        features.push_back(representations.back());
+        auto test_representation = features.get_feature_matrix_dense();
+
+        auto n_feature{test_representation.rows()};
+        auto n_center{test_representation.cols()};
+        BOOST_CHECK_EQUAL(ref_representation.size(), n_feature);
+        for (size_t row_i{0}; row_i < ref_representation.size(); row_i++) {
+          BOOST_CHECK_EQUAL(ref_representation[row_i].size(), n_center);
+          for (size_t col_i{0}; col_i < ref_representation[row_i].size();
+               ++col_i) {
+            auto diff{std::abs(ref_representation[row_i][col_i] -
+                               test_representation(row_i, col_i))};
+            BOOST_CHECK_LE(diff, 1e-12);
           }
         }
       }
+      manager_i += 1;
     }
   }
 
   BOOST_AUTO_TEST_SUITE_END();
 
+  // /* ----------------------------------------------------------------------
```

Comment:
What's the purpose of all this commented stuff?

Reply:
forgot to delete it.